grapple/types/redirects.py

```diff
@@ -1,38 +1,72 @@
+import copy
+
+from typing import Optional
+
 import graphene
 
-from django.conf import settings
 from wagtail.contrib.redirects.models import Redirect
+from wagtail.models import Page, Site
+
+from grapple.types.sites import SiteObjectType
 
 from .pages import get_page_interface
 
 
-class RedirectType(graphene.ObjectType):
+class RedirectObjectType(graphene.ObjectType):
     old_path = graphene.String(required=True)
     old_url = graphene.String(required=True)
-    new_url = graphene.String(required=True)
+    new_url = graphene.String(required=False)
     page = graphene.Field(get_page_interface())
+    site = graphene.Field(SiteObjectType, required=True)
     is_permanent = graphene.Boolean(required=True)
 
-    # Give old_path with BASE_URL attached.
-    def resolve_old_url(self, info, **kwargs):
-        return settings.BASE_URL + self.old_path
+    def resolve_old_url(self, info, **kwargs) -> str:
+        """
+        Resolve the value of `old_url` using the `root_url` of the associated
+        site and `old_path`.
+        """
 
-    # Get new url
-    def resolve_new_url(self, info, **kwargs):
-        if self.redirect_page is None:
-            return self.link
+        return f"{self.site.root_url}/{self.old_path}"
```

Comment:
**Issue:** This always results in a double slash after the hostname because `old_path` enforces that the path begins with `/`. 

This is tested using Wagtail v5.1.2, so may not be true for other versions.

Can you check? We may be able to change this to:

```suggestion
        return self.site.root_url + self.old_path
```

Reply:
I was using Wagtail v5.2.1. I see the `normalize_path` function in the Redirect model, and it should be adding that `/`, but when I run tests the `old_url` is generated in the correct way.

Reply:
Addressed this in 6950794a844166aa5bb9300f3a51ea3ea6b9143b.